aten/src/ATen/native/ForeachUtils.h

```diff
@@ -252,10 +252,16 @@ using IndicesT = std::vector<size_t>;
 using nested_optional_tensorvec_t =
     std::vector<std::vector<c10::optional<at::Tensor>>>;
 using TensorsAndIndicesT = std::pair<nested_optional_tensorvec_t, IndicesT>;
-using FlatMap = std::unordered_map<
-    DeviceDtypeKey,
-    TensorsAndIndicesT,
-    ParamsHash<DeviceDtypeKey>>;
+
+struct _DeviceDtypeHasher {
+  std::size_t operator()(const DeviceDtypeKey& k) const noexcept {
+    static std::hash<at::Device> device_hasher;
+    static std::hash<at::ScalarType> scalar_type_hasher;
+    return device_hasher(k.first) ^ scalar_type_hasher(k.second);
```

Comment:
Would it be clearer to use 
```
std::hash(k.first) ^ std::hast(k.second)
```
?

Reply:
Sure, this is just debugging code for now. Once the pipeline is green, I will clean up the code and re-add the checks I removed.